### PR TITLE
tests: fix expect tests and add EOF linter

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

While looking why https://github.com/algorand/go-algorand/pull/6089 expect test did not work as expected, I found it copied the `expect` statement from the `CheckNetworkAddressForCors` function, and it does not perform check correctly ignoring the fact of missing expected output. 
I fixed that by added `eof` sub statement which is hit when no other statement executed previously and when there is no `exp_continue`. After that the test started failing since it was checking KMD headers without setting `allowed_origins` configuration option.

@cce also discovered the same issue in some other tests and added a linter.

## Test Plan

This is a test fix.

### CODE REVIEW

**Review by gmalouf — APPROVED** [review]

File: test/e2e-go/cli/goal/expect/goalExpectCommon.exp [hunk 1]
```diff
@@ -726,7 +742,7 @@ proc ::AlgorandGoal::SignTransaction { WALLET_NAME WALLET_PASSWORD INPUT_TXN OUT
         expect {
             timeout { ::AlgorandGoal::Abort "Timed out signing transaction"  }
 	        "Please enter the password for wallet '$WALLET_NAME':" { send "$WALLET_PASSWORD\r" ; exp_continue}
-	        eof
+	        eof { ::AlgorandGoal::Abort "EOF signing transaction"  }
```
Comment (algorandskiy) [inline test/e2e-go/cli/goal/expect/goalExpectCommon.exp:745]: `exp_continue` causes `eof` to be called as well so this test most likely will fail

**Review by algorandskiy — COMMENTED** [review]
some `eof`s looks exceeding because of variables checking after the `expect` statement but does not hurt

File: test/e2e-go/cli/goal/expect/goalExpectCommon.exp [hunk 2]
```diff
@@ -565,7 +578,7 @@ proc ::AlgorandGoal::AssetCreate { CREATOR WALLET_NAME WALLET_PASSWORD TOTAL_SUP
         expect {
             timeout { ::AlgorandGoal::Abort "Timed out create asset"  }
 	        "Please enter the password for wallet '$WALLET_NAME':" { send "$WALLET_PASSWORD\r"; exp_continue }
-	        eof
+	        eof { ::AlgorandGoal::Abort "EOF create asset"  }
```
Comment (algorandskiy) [inline test/e2e-go/cli/goal/expect/goalExpectCommon.exp:581]: the same here, `exp_continue` causes `eof` an the test will fail

File: test/e2e-go/cli/goal/expect/tealConsensusTest.exp [hunk 3]
```diff
@@ -84,16 +88,18 @@ if { [catch {
         " - pass -" { puts "small-sig dryrun pass" }
         "REJECT" { ::AlgorandGoal::Abort $expect_out(buffer) }
         "static cost budget" { ::AlgorandGoal::Abort $expect_out(buffer) }
+        eof { ::AlgorandGoal::Abort $expect_out(buffer) }
     }
 
-    teal "$TEST_ROOT_DIR/slow-sig.teal" 2 1 20001
+    teal "$TEST_ROOT_DIR/slow-sig.teal" 4 1 20001
```
Comment (algorandskiy) [inline test/e2e-go/cli/goal/expect/tealConsensusTest.exp:94]: see the original [PR](https://github.com/algorand/go-algorand/pull/2126/files#diff-9d9a133c6ac373272816e6541cd66e3238317f403f743da0215d489f25341025) that added teal v4 with back jumps and subroutines so check does not enforce cost budget

**Review by gmalouf — COMMENTED** [review]
Added 1 ask in Makefile - is the idea that eof wasn't being called in some corner cases/we ended up with false positives?

File: Makefile [hunk 4]
```diff
@@ -114,6 +114,9 @@ fix: build
 lint: deps
 	$(GOPATH1)/bin/golangci-lint run -c .golangci.yml
 
+expectlint:
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply Makefile:117]: I've never run sanity before, and I've tried fix right now and it takes forever to complete, so I guess nobody runs it as well?

[the remark this answers is not in the document]
Comment (gmalouf) [reply Makefile:117]: I run it before pushing most of the time. Codecoverage on github will catch though.

**Review by gmalouf — APPROVED** [review]

**Pull request merged**